src/wasm-builder.h

```diff
@@ -35,10 +35,12 @@ struct NameType {
 
 class Builder {
   MixedArena& allocator;
+  Module& wasm;
 
 public:
-  Builder(MixedArena& allocator) : allocator(allocator) {}
-  Builder(Module& wasm) : allocator(wasm.allocator) {}
+  Builder(MixedArena& allocator, Module& wasm)
+    : allocator(allocator), wasm(wasm) {}
```

Comment:
As a followup I think we can remove this constructor. The allocator should always be from the wasm, so there is no need to pass an allocator manually.

Reply:
I did see at least one use where it seemed that a single allocator was used for potentially processing multiple modules, and I didn't know how intentional that was, so I left it in.